spec/datadog/appsec/metrics/telemetry_spec.rb

```diff
@@ -93,4 +93,139 @@
       end
     end
   end
+
+  describe '.report_waf_requests' do
+    context 'when waf metrics have at least one match' do
+      let(:waf_metrics) do
+        Datadog::AppSec::Metrics::Collector::Store.new(
+          evals: 10, matches: 1, errors: 0, timeouts: 0, duration_ns: 100_000, duration_ext_ns: 50_000
+        )
+      end
+
+      let(:request_milestones) { Datadog::AppSec::RequestMilestones.new }
+
+      it 'reports waf.requests telemetry metric' do
+        expect(telemetry).to receive(:inc).with(
+          'specsec', 'waf.requests', 1, tags: {
+            waf_version: '1.42.99',
+            rule_triggered: 'true',
+            request_blocked: 'false',
+            rate_limited: 'false',
+            waf_error: 'false',
+            waf_timeout: 'false'
+          }
+        )
+
+        described_class.report_waf_requests(waf_metrics, request_milestones)
+      end
+    end
+
+    context 'when waf metrics have at least one error' do
+      let(:waf_metrics) do
+        Datadog::AppSec::Metrics::Collector::Store.new(
+          evals: 10, matches: 0, errors: 1, timeouts: 0, duration_ns: 100_000, duration_ext_ns: 50_000
+        )
+      end
+
+      let(:request_milestones) { Datadog::AppSec::RequestMilestones.new }
+
+      it 'reports waf.requests telemetry metric' do
+        expect(telemetry).to receive(:inc).with(
+          'specsec', 'waf.requests', 1, tags: {
+            waf_version: '1.42.99',
+            rule_triggered: 'false',
+            request_blocked: 'false',
+            rate_limited: 'false',
+            waf_error: 'true',
+            waf_timeout: 'false'
+          }
+        )
```

Comment:
I think to make emphasis on what we actually testing here it makes sense to decompose expectation and limit it to the actual check of error tag only.

Yes we probably want to have a single test to check for all tags, but all the similarly looking tests of specific tags, should just check those tags only, like

```ruby
expect(telemetry).to receive(:inc).with(
  'specsec', 'waf.requests', 1, tags: hash_including(waf_error: 'true')
)
```